storage/innobase/handler/i_s.cc

```diff
@@ -6823,10 +6861,13 @@ i_s_sys_tablespaces_fill_table(
 	mutex_exit(&dict_sys.mutex);
 	mem_heap_free(heap);
 
-	i_s_dict_fill_sys_tablespaces(
+	if (int err = i_s_dict_fill_sys_tablespaces(
 		thd, fil_system.temp_space->id,
 		fil_system.temp_space->name,
-		fil_system.temp_space->flags, tables->table);
+		fil_system.temp_space->flags, tables->table)) {
+		ut_free(pcur.old_rec_buf);
+		DBUG_RETURN(err);
+	}
 
 	DBUG_RETURN(0);
```

Comment:
Is the `ut_free()` really needed? We do not invoke it in the normal case. When would the cursor position be saved? Could the `ut_free()` be needed also in the normal case?

The `if` around `int err` could be removed here. We could either just `DBUG_RETURN(i_s_dict_fill_sys_tablespaces(…))` or execute the `if` body unconditionally.